presto-iceberg/src/test/java/com/facebook/presto/iceberg/IcebergDistributedSmokeTestBase.java

```diff
@@ -1031,6 +1031,30 @@ public void testCreateNestedPartitionedTable(Session session, FileFormat fileFor
         dropTable(session, "test_nested_table2");
     }
 
+    @Test
+    public void testPartitionedByTime()
+    {
+        testSelectOrPartitionedByTime(getSession(), false, FileFormat.PARQUET);
+        testSelectOrPartitionedByTime(getSession(), false, FileFormat.ORC);
+        testSelectOrPartitionedByTime(getSession(), true, FileFormat.PARQUET);
+        testSelectOrPartitionedByTime(getSession(), true, FileFormat.ORC);
+    }
+
+    private void testSelectOrPartitionedByTime(Session session, boolean partitioned, FileFormat format)
+    {
+        String tableName = format("test_%s_by_time", partitioned ? "partitioned" : "selected");
+        String partitioning = partitioned ? ", partitioning = ARRAY['x']" : "";
+        assertUpdate(format("CREATE TABLE %s (x TIME, y BIGINT) WITH (format = '%s'%s)", tableName, format, partitioning));
+        assertUpdate(format("INSERT INTO %s VALUES (TIME '10:12:34', 12345)", tableName), 1);
+        assertQuery(format("SELECT COUNT(*) FROM %s", tableName), "SELECT 1");
+        assertQuery(format("SELECT x FROM %s", tableName), "SELECT CAST('10:12:34' AS TIME)");
+        assertUpdate(format("INSERT INTO %s VALUES (TIME '9:00:00', 67890)", tableName), 1);
+        assertQuery(format("SELECT COUNT(*) FROM %s", tableName), "SELECT 2");
```

Comment:
How is this test and `assertQuery(format("SELECT COUNT(*) FROM %s", tableName), "SELECT 1");` related to the issue?

Reply:
I suppose this is just another assertion that what's happening is what we think is happening, I can remove it if you think it's unnecessary.

Reply:
Understood, lets keep it.